webapp/graphite/storage.py

```diff
@@ -77,18 +79,42 @@ def pool_exec(self, jobs, timeout):
 
     def fetch(self, patterns, startTime, endTime, now, requestContext):
         # deduplicate patterns
-        patterns = list(set(patterns))
+        patterns = sorted(set(patterns))
 
         if not patterns:
             return []
 
         log.debug(
             'graphite.storage.Store.fetch :: Starting fetch on all backends')
 
-        jobs = [
-            Job(finder.fetch, patterns, startTime, endTime, now=now, requestContext=requestContext)
-            for finder in self.get_finders(requestContext.get('localOnly'))
-        ]
+        jobs = []
+        tag_patterns = None
+        pattern_aliases = defaultdict(list)
+        for finder in self.get_finders(requestContext.get('localOnly')):
+          if getattr(finder, 'tags', False):
+            job_patterns = patterns
```

Comment:
maybe this could be a continue here to avoid the else going to much to the right ?